docs/package_templates/header_only/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.51.3"
+
+
+class PackageConan(ConanFile):
+    name = "package"
+    description = "short description"
+    license = "" # Use short name only, conform to SPDX License List: https://spdx.org/licenses/
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/project/package"
+    topics = ("topic1", "topic2", "topic3", "header-only") # no "conan"  and project name in topics, keep 'header-only'
+    settings = "os", "arch", "compiler", "build_type" # even for header only
+    no_copy_source = True # do not copy sources to build folder for header only projects, unless, need to apply patches
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "14.1",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # prefer self.requires method instead of requires attribute
+        self.requires("dependency/0.8.1")
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        # compiler subsettings are not available when building with self.info.clear()
+        if self.info.settings.get_safe("compiler.cppstd"):
```

Comment:
So it doesn't make sense to test this in validate() since it will never enter into this branch. And for me it looks like all this new validate logic based on self.info in conan v2 is broken?

Reply:
No workaround. Need to update by using `self.settings` unfortunately. So far, there is no solution: https://github.com/conan-io/docs/pull/2778